src/nvim/eval.c

```diff
@@ -17289,7 +17289,7 @@ static bool write_list(FileDescriptor *const fp, const list_T *const list,
       }
     }
   }
-  if ((error = file_fsync(fp)) != 0) {
+  if ((error = file_flush(fp)) != 0) {
```

Comment:
This guarantees that buffer contents will be sent to the OS (write() syscall) and buffer will be cleared, nothing more.